scripts/test/Muon/grouping_tab/pairing_table_group_selector_test.py

```diff
@@ -0,0 +1,164 @@
+        self.assertEqual(self.get_group_2_selector(0).currentText(), "my_group_1")
+
+    def test_that_added_groups_appear_in_group_combo_boxes_in_new_pairs(self):
+        self.add_three_groups_to_model()
+        self.presenter.handle_add_pair_button_clicked()
+
+        self.assertEqual(self.get_group_1_selector(0).count(), 3)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_0"), -1)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_1"), -1)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_2"), -1)
+
+    def test_that_get_index_of_text_returns_correct_index_if_text_exists(self):
+        self.add_three_groups_to_model()
+        self.presenter.handle_add_pair_button_clicked()
+
+        index = self.view.get_index_of_text(self.get_group_1_selector(0), 'my_group_1')
+
+        self.assertEqual(index, 1)
+
+    def test_that_get_index_of_text_returns_0_if_text_does_not_exists(self):
+        self.add_three_groups_to_model()
+        self.presenter.handle_add_pair_button_clicked()
+
+        index = self.view.get_index_of_text(self.get_group_1_selector(0), 'random string')
+
+        self.assertEqual(index, 0)
+
+    def test_that_added_groups_appear_in_group_combo_boxes_in_existing_pairs_if_update_called(self):
+        self.presenter.handle_add_pair_button_clicked()
+        self.add_three_groups_to_model()
+        # the following method must be called
+        self.presenter.update_view_from_model()
+
+        self.assertEqual(self.get_group_1_selector(0).count(), 3)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_0"), -1)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_1"), -1)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_2"), -1)
+
+    def test_that_changing_group_selection_triggers_cell_changed_method_in_view(self):
+        self.add_three_groups_to_model()
+        self.presenter.handle_add_pair_button_clicked()
+        self.presenter.handle_add_pair_button_clicked()
+
+        self.view.on_cell_changed = mock.Mock()
+        self.get_group_1_selector(0).setCurrentIndex(1)
+
+        self.assertEqual(self.view.on_cell_changed.call_count, 1)
+        self.assertEqual(self.view.on_cell_changed.call_args_list[0][0], (0, 1))
+
+    def test_that_removing_groups_and_then_calling_update_removes_groups_from_selections(self):
+        self.add_three_groups_to_model()
+        self.presenter.handle_add_pair_button_clicked()
+        del self.data.groups["my_group_1"]
+        del self.data.groups["my_group_2"]
+        self.presenter.update_view_from_model()
+
+        self.assertEqual(self.get_group_1_selector(0).count(), 1)
+        self.assertEqual(self.get_group_2_selector(0).count(), 1)
+        self.assertNotEqual(self.get_group_1_selector(0).findText("my_group_0"), -1)
+        self.assertNotEqual(self.get_group_2_selector(0).findText("my_group_0"), -1)
+
```

Comment:
should we also check:

1. if a pair is set up and a new group is added, then pair's groups have not changed
2. if a pair is set up and one of the groups is removed it is handled gracefully.